tests/llama_index/test_llama_index_tracer.py

```diff
@@ -122,20 +198,19 @@ def test_trace_llm_error(monkeypatch):
         timeout=1,
         max_retries=0,
     )
+    message = ChatMessage(role="system", content="Hello")
 
     with pytest.raises(openai.APIConnectionError, match="Connection error."):
-        llm.chat([ChatMessage(role="system", content="Hello")])
+        _ = list(llm.stream_chat([message])) if is_stream else llm.chat([message])
```

Comment:
```suggestion
        if is_stream:
            next(llm.stream_chat([message])):
        else:
            llm.chat([message])
```

I'd use a if statement here since we don't use the return value.

Reply:
Oops, actually ruff complains about having `if-else` within `pytest.raises`.
> tests/llama_index/test_llama_index_tracer.py:201:5: PT012 `pytest.raises()` block should contain a single simple statement

Lemme keep the original statement for now

Reply:
This could be a bug in ruff.

Reply:
It actually seems to be the intention of rule. But I feel like the rule is too rigorous and cause false positives🤔
https://github.com/m-burst/flake8-pytest-style/blob/master/docs/rules/PT012.md
Do you think we can just disable it globally?

Reply:
@B-Step62 I'd keep it to prevent code like this:

```
with pytest.raises(...):
   should_not_raise()  # but raises
   should_raise()  # not executed
```

Reply:
Ok, I'll just to noqa then